### PR TITLE
feat/python: Str to int

### LINKED ISSUE #963: Replace unixTime with slotIndex in Timelock/Expiration - python
https://github.com/iotaledger/tips-draft/blob/tip38/tips/TIP-0038/tip-0038.md#timelock-unlock-condition
https://github.com/iotaledger/tips-draft/blob/tip38/tips/TIP-0038/tip-0038.md#expiration-unlock-condition

### LINKED ISSUE #1163: Python use int for u64 values
## Description

Python use int for u64 and u128 values https://github.com/iotaledger/iota-sdk/pull/1152#discussion_r1317234431
So change serialization to convert the types

Can probably use something like this for affected fields
```
 = field(metadata=config(
                      decoder=int_from_str
                      encoder=int_to_str
                  ))
```

## Motivation

Usability 

## Requirements

1. Update types and add custom de/serialization for the affected fields

## Are you planning to do it yourself in a pull request?

No.

### PR DESCRIPTION
**Pull request by kwek20**
# Description of change

added custom encoder to translate the int to string.
setting type as int already decodes automatically 🎉

## Links to any relevant issues

fixes #1163

also fixes [#963](https://github.com/iotaledger/iota-sdk/issues/963)

## Type of change

- Bug fix (a non-breaking change which fixes an issue)
- Enhancement (a non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Documentation Fix

## How the change has been tested

Added a test_serde for specific serialize tests
ran a few examples which were broken

### CODE REVIEW

File: bindings/python/iota_sdk/types/node_info.py [hunk 1]
```diff
@@ -203,18 +224,34 @@ class ProtocolParameters:
...
+    staking_unbonding_period: int = field(metadata=config(
+        encoder=str
+    ))
     validation_blocks_per_slot: int
-    punishment_epochs: str
-    liveness_threshold: str
-    min_committable_age: str
-    max_committable_age: str
-    epoch_nearing_threshold: str
+    punishment_epochs: int = field(metadata=config(
```
Comment (Thoralf-M) [inline bindings/python/iota_sdk/types/node_info.py]: Now a u32, so not str encoding needed

File: bindings/python/iota_sdk/types/node_info.py [hunk 2]
```diff
@@ -203,18 +224,34 @@ class ProtocolParameters:
...
+        encoder=str
+    ))
+    liveness_threshold: int = field(metadata=config(
```
Comment (Thoralf-M) [inline bindings/python/iota_sdk/types/node_info.py]: Now u16, no str encoding needed

File: bindings/python/tests/test_serde.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        "timelockUnixTime": "2"
+    }
+
+    unlocks = Unlocks.from_dict(unlocks_dict)
+    assert unlocks.expiration_unix_time == 1
+    assert unlocks.timelock_unix_time == 2
+    assert unlocks.to_dict() == unlocks_dict
+
+    opt_unlocks_dict = {
+        "expirationUnixTime": "1"
```
Comment (Thoralf-M) [inline bindings/python/tests/test_serde.py]: No unix time anymore

**Review by Thoralf-M — APPROVED** [review]

**Review by DaughterOfMars — COMMENTED** [review]
does the encoder also handle decode?

**Comment by kwek20** [comment]
> does the encoder also handle decode?

yea, "setting type as int already decodes automatically 🎉"

**Review by DaughterOfMars — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**